src/librustc_span/lib.rs

```diff
@@ -1651,3 +1655,9 @@ where
         }
     }
 }
+
+impl<HirCtx: crate::HashStableContext> HashStable<HirCtx> for DefId {
+    fn hash_stable(&self, hcx: &mut HirCtx, hasher: &mut StableHasher) {
+        hcx.hash_def_id(*self, hasher)
+    }
+}
```

Comment:
Could you move this impl inside `def_id.rs` so the file could be easily moved to its old place or some other place later.